### DISCUSSION TITLE
Client behaviour when enabling IPv6 and using SRV records that only resolve to legacy IP addresses.

### DISCUSSION THREAD

**Discussion by rdica** [jamulussoftware/jamulus#3178] [Bug Investigation]
Hi all, Jamulus 3.10.0 client and server.

I'm not sure this is a bug, or intended behaviour. I'm aware IPv6 support, while mostly working, may still have some issues.
I found one that seems to be related to using SRV DNS records.

I was playing around with SRV records, pointing some to my pub/priv servers on hosts that are dual stack.
Each host has its own A and AAAA records.

I start my client with `-6` arg, and it connects fine using the SRV records. Connections are established over IPv6.

I have one server listening on legacy IP only. I created an SRV record for it and tried connecting my client via the SRV record but no connect attempt is made. If I start the client without `-6`, it will resolve the SRV and connect.

It appears adding `-6` forces an IPv6 only resolution attempt and if no AAAA record is found for the destination in the SRV, the client doesn't have anything to connect to.

If I connect to the FQDN (A record) itself instead of using SRV, the client connects fine even though `-6` is enabled. This behaviour is expected. I expected the same behaviour when accessing via SRV but apparently this isn't the case.

Thoughts?

**Comment by rdica** [comment]
My apologies, please disregard report. I discovered I hadn't removed the AAAA record for the host in question even though I don't run jamulus server with `-6` on that host. Removing the record leaves only the legacy IP and the client connects fine using the SRV when starting it with `-6`, the behaviour I expected.

User error, my apologies.

**Reply by pljones** [reply]
Thanks for raising and posting the resolution.  It's especially interesting to see the interaction with the new SRV feature.

**Reply by rdica** [reply]
I did find another anomaly, maybe this is a Happy Eyeballs thing.
On a dual stack host where jamulus server itself is not listening on IPv6, client connections don't fail over to legacy IP. It appears that if the client cannot connect to the IPv6 address:port, no attempt is made to try legacy address:port afterwards.
